pymeasure/instruments/newport/esp300.py

```diff
@@ -176,7 +175,7 @@ class Axis(object):
         "MD?",
         """ Returns a boolean that is True if the motion is finished.
         """,
-        cast=bool
```

Comment:
The old version uses the casting in the adapter class. I would prefer to keep this behavior.
https://github.com/ralph-group/pymeasure/blob/29ce412f058607075d8d5ad8b3119c40c6f1c819/pymeasure/adapters/adapter.py#L62

Reply:
Hmm, I changed this long enough ago that I can't remember exactly why I did it, but presumably it wasn't returning what I expected. I have a feeling that it would always return `True` before, but I'll check it out again.

Reply:
Interesting. Please verify with the instrument. It would be nice to fix it in the adapter class, since we use this `cast` argument in other contexts.

Reply:
I'm not sure where my other comment went, but I have tested putting `cast = lambda x: bool(float(x))` in thand it does work with the instrument. 
How were you thinking of fixing this in the adapter class? I would think we would need to assume that we would need to always cast to float or something first, but is it reasonable to assume that we are getting numbers in strings from all instruments?

Reply:
I realize the issue now. By default, all returned values are cast as floats. So when `cast=bool` is used, it directly casts as a boolean -- this will return True for strings as long as they are not blank.
```python
>>> s = '1'
>>> bool(s)
True
>>> bool(float(s))
True
>>> s = '0'
>>> bool(s)
True
>>> bool(float(s))
False
>>> 
```
So, we can either:
1. Use `bool(float(x))` in the adapter if `bool` is passed as a cast function.
2. Use `get_process=bool`, since `cast=float` by default.

What do you think?

Reply:
Using `bool(float(x))` in the adapter whenever `bool` is passed as a cast would probably be fine, and would probably be easier to work with for any future instrument development.  Off the top of my head, I can't think of any cases where this would give an unexpected result. If the instrument is returning something strange which should be converted to a boolean (I'm thinking like a string `"true"` or `"T"` or something) then anyone creating a new instrument would probably realize they would need to handle that by passing a `get_process`

Reply:
I've made the change to the base `adapter` class. I tested using `cast == bool` by making a function `def afunc(x): return x == bool` in an interactive environment, and it behaved as expected.